entry/src/entry.rs

```diff
@@ -485,55 +485,86 @@ pub fn start_verify_transactions(
                     gpu_verify_duration_us: 0,
                 });
             }
-            let mut packets = Packets::new_with_recycler(
-                verify_recyclers.packet_recycler,
-                num_transactions,
-                "entry-sig-verify",
-            );
-            // We use set_len here instead of resize(num_transactions, Packet::default()), to save
-            // memory bandwidth and avoid writing a large amount of data that will be overwritten
-            // immediately afterwards. As well, Packet::default() actually leaves the packet data
-            // uninitialized anyway, so the initilization would simply write junk into
-            // the vector anyway.
-            unsafe {
-                packets.packets.set_len(num_transactions);
+            // We purposely use this method to calculate num_vecs instead of
+            // (num_transactions + PACKETS_PER_BATCH - 1) / PACKETS_PER_BATCH
+            // to avoid the possibility of overflow. With this method, it should
+            // not be possible to have enough elements in the virtual address space
+            // to have an overflow in the calculation (and using saturating_add would lead to
+            // an incorrect result in cases where an overflow would happen)
```

Comment:
You could use `checked_add` to protect against this and return an error on overflow.